### PR TITLE
Promote vsphereproviderconfig api to root folder

### PR DESCRIPTION
**Pull request by figo**
1) move API definition to repo root repo by following convention.
2) add vm_folder support in machineConfigure, so that user can specify vm_folder.

/cc @sflxn @sidharthsurana @frapposelli

### CODE REVIEW

**Comment by sflxn** [comment]
Let's move the apis folder into a pkg folder.

File: cloud/vsphere/provisioner/govmomi/create.go [hunk 1]
```diff
@@ -257,33 +258,30 @@ func Properties(vm *object.VirtualMachine) (*mo.VirtualMachine, error) {
...
+func (vc *Provisioner) updateMachineProviderStatus(machine *clusterv1.Machine, field string, value string) error {
 	oldProviderStatus, err := vsphereutils.GetMachineProviderStatus(machine)
 	if err != nil {
 		return err
 	}
 
-	if oldProviderStatus != nil && oldProviderStatus.MachineRef == vmref {
+	oldField := reflect.ValueOf(oldProviderStatus).Elem().FieldByName(field)
+	if !oldField.IsValid() {
+		return errors.New("not a valid field")
```
[the remark this answers is not in the document]
Comment (sidharthsurana) [reply cloud/vsphere/provisioner/govmomi/create.go]: agreed! this message could be improved

**Comment by sidharthsurana** [comment]
lgtm

**Comment by frapposelli** [comment]
Please rebase the PR 🙏

**Comment by figo** [comment]
working on rebase right now, thanks

**Pull request merged**